readme.md

```diff
@@ -6,67 +6,121 @@
+-------------------
+* Getting Started
+    * [Requirements](#requirements)
+    * [Targets](#targets)
+    * [Configuration and Presets](#configuration-and-presets)
+    * [Building from Command Line](#building-from-command-line)
+    * [Building in Visual Studio](#building-in-visual-studio)
     * [Definitions](#definitions)
-* [Overview](docs/Overview.md)
-* [Creating a Project](docs/CreatingAProject.md)
-* [NcEngine Components](docs/EngineComponents.md)
-
-
-## Building
+* More Information
+    * [Overview](docs/Overview.md)
+    * [Creating a Project](docs/CreatingAProject.md)
+    * [NcEngine Components](docs/EngineComponents.md)
+
+### Requirements
+----------------
+* Windows
+* Vulkan SDK
+* CMake 3.20
+* Compiler with c++20 support:
+    * Visual Studio 17 2022
+    * MinGW-w64 9.0.0
+
+> When cloning, use `--recurse-submodules`. If any subdirectories of nc/external are empty, the repo was cloned non-recursively. Use `git submodule update --init` to get the dependencies.
+
+### Targets
 -----------
-Current limitations:
-* NcEngine depends on Windows. There will likely be cross-platform support in the future, but it is not actively being pursued.
-* There are a few places where gcc specific code hasn't been fixed. This unfortunately means that mingw is the only currently supported compiler. This is planned to be fixed in the near future.
+* NcEngine: Core Library [Required]
+    * Type: Static Library
+    * Directory: nc
+* NcEditor: Application for project creation and management.
+    * Type: Executable
+    * Directory: editor
+    * Dependencies: NcEngine [with -DNC_EDITOR_ENABLED=ON]
+* Sample: Application containing demo, test, and benchmark scenes.
+    * Type: Executable
+    * Directory: project
+    * Dependencies: NcEngine
+* Tests:
+    * Type: Executable(s)
+    * Directory: test
+
+NcEngine is the library games link against. NcEditor and the sample application are optional and depend on NcEngine being built *and* installed.
+
+### Configuration and Presets
+-----------------------------
+Each target's root directory contains a CMakePresets.json file with common configuration and build presets for both Visual Studio and Ninja + MinGW. The available presets can be viewed with `cmake -S <target-source-dir> --list-presets`.
+
+Each of these presets follows the naming scheme `<target>-<generator>-<configuration>-[modifier]`, where modifier describes optional features to include:
+* WithEditor: Include editor-specific code.
+* WithProfiling: Include Optick profiling code.
+* WithValidation: Enable Vulkan validation layers.
+
+Each preset provides a different set of [definitions](#definitions). When adding custom presets to CMakeUserPresets.json, NC_INSTALL_DIR must be defined.
+
+More on CMake Presets:
+* [Documentation](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html)
+* [CMake Presets in Visual Studio](https://docs.microsoft.com/en-us/cpp/build/cmake-presets-vs?view=msvc-170)
+
+### Building from Command Line
```

Comment:
I would move "Building from Command Line" to underneath "Building in Visual Studio" - the Visual studio instructions are simpler, and its also more common for windows devs to use VS. It provides a nice break in the detail and a good foundation for moving on with the rest of the readme